env.sh

```diff
@@ -21,55 +21,84 @@ run_cmd() {
+      run_cmd rm -rf $CMSSW_VER
     fi
-  elif [[ $MODE = "conda" ]]; then
+    echo "Creating $CMSSW_VER area for CentOS$os_version in $PWD ..."
+    run_cmd scramv1 project CMSSW $CMSSW_VER
+    run_cmd cd $CMSSW_VER/src
+    run_cmd eval `scramv1 runtime -sh`
+    run_cmd mkdir TauMLTools
+    run_cmd cd TauMLTools
+    run_cmd ln -s "$TAU_ML_DIR/Analysis" Analysis
+    run_cmd ln -s "$TAU_ML_DIR/Core" Core
+    run_cmd ln -s "$TAU_ML_DIR/Production" Production
+    run_cmd scram b -j8
+    run_cmd touch "$this_dir/soft/CentOS$os_version/$CMSSW_VER/.installed"
+  fi
+}
+
+install_cmssw() {
+  local this_file="$( [ ! -z "$ZSH_VERSION" ] && echo "${(%):-%x}" || echo "${BASH_SOURCE[0]}" )"
+  local this_dir="$( cd "$( dirname "$this_file" )" && pwd )"
+  local scram_arch=$1
+  local cmssw_version=$2
+  local os_version=$3
+  if [[ $os_version < 8 ]] ; then
+    local env_cmd=cmssw-cc$os_version
+  else
+    local env_cmd=cmssw-el$os_version
+  fi
+  if ! [ -f "$this_dir/soft/CentOS$os_version/$CMSSW_VER/.installed" ]; then
+    run_cmd $env_cmd --command-to-run /usr/bin/env -i HOME=$HOME bash "$this_file" install_cmssw $scram_arch $cmssw_version $os_version
+  fi
+}
+
+action() {
+  local this_file="$( [ ! -z "$ZSH_VERSION" ] && echo "${(%):-%x}" || echo "${BASH_SOURCE[0]}" )"
+  local this_dir="$( cd "$( dirname "$this_file" )" && pwd )"
+  local MODE=$1
+
+  export ANALYSIS_PATH="$this_dir"
+  export ANALYSIS_DATA_PATH="$ANALYSIS_PATH/data"
+  export X509_USER_PROXY="$ANALYSIS_DATA_PATH/voms.proxy"
+
+  export PYTHONPATH="$this_dir:$PYTHONPATH"
+  export LAW_HOME="$this_dir/.law"
+  export LAW_CONFIG_FILE="$this_dir/law/law.cfg"
+
+  run_cmd mkdir -p "$ANALYSIS_DATA_PATH"
+
+  local os_version=$(cat /etc/os-release | grep VERSION_ID | sed -E 's/VERSION_ID="([0-9]+)"/\1/')
+  local default_cmssw_ver=CMSSW_12_4_10
+  export DEFAULT_CMSSW_BASE="$ANALYSIS_PATH/soft/CentOS$os_version/$default_cmssw_ver"
+
+  if [[ $MODE = *"cmssw"* ]]; then
+    run_cmd install_cmssw slc7_amd64_gcc10 $default_cmssw_ver 7
+    run_cmd install_cmssw el8_amd64_gcc10 $default_cmssw_ver 8
+
+    #for phase2
+    run_cmd install_cmssw slc7_amd64_gcc900 CMSSW_11_2_5 7
+    run_cmd install_cmssw slc7_amd64_gcc900 CMSSW_11_3_0 7
+
+    alias cmsEnv="env -i HOME=$HOME ANALYSIS_PATH=$ANALYSIS_PATH ANALYSIS_DATA_PATH=$ANALYSIS_DATA_PATH X509_USER_PROXY=$X509_USER_PROXY DEFAULT_CMSSW_BASE=$DEFAULT_CMSSW_BASE $ANALYSIS_PATH/RunKit/cmsEnv.sh"
+    alias cmsEnv11_2="env -i HOME=$HOME ANALYSIS_PATH=$ANALYSIS_PATH ANALYSIS_DATA_PATH=$ANALYSIS_DATA_PATH X509_USER_PROXY=$X509_USER_PROXY DEFAULT_CMSSW_BASE=$ANALYSIS_PATH/soft/CentOS$os_version/CMSSW_11_2_5 $ANALYSIS_PATH/RunKit/cmsEnv.sh"
+    alias cmsEnv11_3="env -i HOME=$HOME ANALYSIS_PATH=$ANALYSIS_PATH ANALYSIS_DATA_PATH=$ANALYSIS_DATA_PATH X509_USER_PROXY=$X509_USER_PROXY DEFAULT_CMSSW_BASE=$ANALYSIS_PATH/soft/CentOS$os_version/CMSSW_11_3_0 $ANALYSIS_PATH/RunKit/cmsEnv.sh"
+  fi
+
+  if [[ $MODE == *"conda"* ]]; then
```

Comment:
Would it make sense to update to the latest version of `law` in conda? (currently we are with `0.1.5`, but the latest is [`0.1.12`](https://github.com/riga/law/releases/tag/v0.1.12)). But I am not sure about the advantages of this, as well as about the breaking changes.